### ISSUE TITLE
Add component for showing full-width images and figure

### ISSUE THREAD

**Issue by nilsml** [equinor/energyvision#273]
**In order to show icons and illustrations, we will add a component that supports images in different formats**

## Acceptance criteria
1. It supports full-width images with an aspect ratio where Sanity's hotspot features and crop is used
1. It supports ordinary images without aspect ratio (no hotspot and crop) with caption and attributions

## Investigation
1. Need to figure out what the aspect ration should be. Can it be added? The aspect ratio is 10:3

## In order to ensure good quality we should (Keep the ones we need for this US)
- [ ] Perform integration testing
- [ ] Perform functional testing
- [ ] Extra focus on usability
- [ ] Test for portability
- [ ] Test for accessibility
- [ ] Create automated acceptance test/smoke test
- [ ] Perform performance and load test
- [ ] Perform user testing
- [ ] Perform security testing

**Comment by SvSven** [comment]
Need to figure out if this should have a max. height, or whether the crop/original dimensions should be respected.
The problem with the latter is that, because the image should be the full page width, we may end up with a very large image - both in terms of width and height.

**Comment by wenche** [comment]
We have identified the need for two different kind of images:

1. Full width image as in the POC (height is 37.5% of the width). This image is for decorative purposes or for setting a tone/creating a mood. This one will be used with the hotspot features in Sanity

2. An image that preserves the actual height and width from the original image but with a max width. The maximum width will be the same as for the text content in the text block. The hotspot features will not be used.

**Comment by wenche** [comment]
We do have images :thumbsup:

https://energyvision.app.radix.equinor.com/careers at the bottom

![Skjermbilde 2021-06-01 kl. 13.55.45.png](https://images.zenhubusercontent.com/5fb7aa9b653d5135bb8c50f5/1d772c09-6faa-460f-8bdf-293bc71628f9)![Skjermbilde 2021-06-01 kl. 13.56.06.png](https://images.zenhubusercontent.com/5fb7aa9b653d5135bb8c50f5/3bd06900-7b2c-4d18-8c0e-8866b7318122)

In Sanity
![Skjermbilde 2021-06-01 kl. 13.56.46.png](https://images.zenhubusercontent.com/5fb7aa9b653d5135bb8c50f5/f529b9e1-7422-405d-81be-40f7e218ef05)

We will still need #268 and #267 in order to make it easier to understand in Sanity for publishers. The full width image uses hotspot and crop features in the web, the other one doesn't (because then the component will not be usable for things like images of maps)

@BirteThornquist and I will go through the last details tomorrow, _not_ ready for testing yet.

**Comment by wenche** [comment]
Should be ready for testing or verification, but it's mostly reuse of already existing existing code. Se /careers for an example of both full width and image. @petterwildhagen

**Comment by petterwildhagen** [comment]
@wenche the URL .../careers returns a 404

**Comment by meols** [comment]
@SvSven @wenche @BirteThornquist 
One comment below, otherwise good:
- [x ] It supports full-width images with an aspect ratio where Sanity's hotspot features and crop is used
- [x ] It supports ordinary images without aspect ratio (no hotspot and crop) with caption and attributions

How should caption and attribution be placed? On the same line or on separate lines? Currently on the same line

@petterwildhagen

**Comment by wenche** [comment]
@meols  `How should caption and attribution be placed? On the same line or on separate lines? Currently on the same line`

Birte, Nick and I had a discussion about this when we worked with the news articles, and Nick wanted the flexibility with two simple text fields for this. And in the design, they are just placed one one line in the web.

This is easy enough to change to use two lines, use italic for the attribution, always use a punctuation mark after the caption +++ if we want to at a later stage. It's one of those things that is easier to see after we have migrated more content.
